src/main/java/com/epam/javacourses2016/task13/Graph.java

```diff
@@ -0,0 +1,75 @@
+package com.epam.javacourses2016.task13;
+
+import java.util.HashSet;
+import java.util.Set;
+
+
+/**
+ * Created by Рамиль on 29.11.2016.
+ */
+public class Graph extends AbstractGraphCreator.AbstractGraph {
+
+
+    public Set<Node> nodes;
+
+    public Graph(int numberNodes) {
+        super(numberNodes);
+        nodes = new HashSet<>();
+        for (int i = 0; i < NUMBER_NODES; i++) {
+            Node node = new Node(i);
+            nodes.add(node);
+        }
+    }
+
+    @Override
+    public void addEdge(int first, int second) {
+        for (Node node : nodes) {
```

Comment:
Set or Map give us O(logn) or even O(1) (for HashMap or HashSet with no collisions) on searching for element, but your implementations of Graph and Node don't use that speedup using iteration through collection with just O(n) :snail: